NuclearData/ceNeutronData/aceDatabase/aceNeutronNuclide_class.f90

```diff
@@ -589,45 +587,112 @@ end subroutine getUrrXSs
   !! an energy range given by an upper and lower energy bound.
   !!
   !! Args:
-  !!   upperE [in]  -> Upper bound of energy range
-  !!   upperE [in]  -> Upper bound of energy range
+  !!   eLower [in]  -> Lower bound of energy range
+  !!   eUpper [in]  -> Upper bound of energy range
+  !!   MT     [in]  -> MT number of the requested reaction cross section
   !!   maj [out]    -> Maximum scattering cross section within energy range
   !!
-  function elScatteringMaj(self, lowerE, upperE) result (maj)
+  function getMajXS(self, eLower, eUpper, MT) result (maj)
     class(aceNeutronNuclide), intent(in)  :: self
-    real(defReal), intent(in)             :: lowerE
-    real(defReal), intent(in)             :: upperE
+    real(defReal), intent(in)             :: eLower
+    real(defReal), intent(in)             :: eUpper
+    integer(shortInt), intent(in)         :: MT
     real(defReal)                         :: maj
-    integer(shortInt)                     :: idx
+    integer(shortInt)                     :: reaction, idx
     real(defReal)                         :: f, E, xs
+    character(100), parameter :: Here = 'getMajXS (aceNeutronNuclide_class.f90)'
+
+    ! Select desired reaction based on requested MT number
+    select case (MT)
+
+      case (N_TOTAL)
+        reaction = TOTAL_XS
+
+      case (N_N_ELASTIC)
+        reaction = ESCATTER_XS
+
+      case default
+        call fatalError(Here, 'Unsupported MT number requested: '//numToChar(MT))
+
+    end select
 
     ! Search for idx, f, and xs for the lower energy limit
-    call self % search(idx, f, lowerE)
+    call self % search(idx, f, eLower)
 
     ! Conservative: choose the xs at the energy point before the lower energy limit
-    f = 0
-    maj = self % scatterXS(idx, f)
+    maj = self % mainData(reaction, idx)
 
     majorantLoop: do
 
       ! Increase index
       idx = idx + 1
 
       ! Find XS and energy at index
-      xs = self % mainData(ESCATTER_XS, idx)
-      E = self % eGrid(idx)
+      xs = self % mainData(reaction, idx)
+      E  = self % eGrid(idx)
 
       ! Compare cross sections and possibly update majorant
       if (xs > maj) then
```

Comment:
Maybe slightly neater: maj = max(maj,xs)?

Reply:
I don't know how I missed that